daml-lf/engine/src/main/scala/com/digitalasset/daml/lf/engine/Engine.scala

```diff
@@ -69,34 +69,41 @@ final class Engine {
     *      because the caller has not provided a required contract instance or package.
     * </li>
     * </ul>
+    *
+    * This method does NOT perform authorization checks; ResultDone can contain a transaction that's not well-authorized.
     */
-  def submit(cmds: Commands): Result[Transaction.Transaction] = {
+  def submit(submitters: Set[Party], cmds: Commands): Result[Transaction.Transaction] = {
```

Comment:
This is the wrong place to put this her. `cmds` already contains the submitter. In fact, what are the semantics of this?